express/properties/non_scalar/averaged_potential.py

```diff
@@ -0,0 +1,36 @@
+import numpy as np
+from scipy.signal import find_peaks
+from express.parsers.settings import Constant
+from express.properties.non_scalar import NonScalarProperty
+
+
+class AveragedPotential(NonScalarProperty):
+    """
+    Holds planar and macroscopic average of a potential (e.g. electrostatic potential).
+    Additionally, minima are included as they can be of further use (e.g. potential lineup method).
+    """
+
+    def __init__(self, name, parser, *args, **kwargs):
+        super(AveragedPotential, self).__init__(name, parser, *args, **kwargs)
+
+        self.data = self.safely_invoke_parser_method("averaged_quantity")
+        self.data["x"] *= Constant.BOHR  # convert to angstrom
+        self.data["p_x"] *= Constant.RYDBERG  # convert to eV
+        self.data["m_x"] *= Constant.RYDBERG  # convert to eV
+
+        self.peak_prominence = [0.3, None]  # minimum and maximum required prominence (see _find_minima)
```

Comment:
Will this be ever overridden? If yes, should we keep in the data structure?

Reply:
So this controls how we find the minima (i.e. peaks of negated array). I was wondering if we should expose that to the user somehow, but that is so specific which makes it really an expert setting. Perhaps we can find a formula for this value that would correctly identify all minima in most cases. We could include it in the property data structure to document how we determined the minima. In any case, I think we should include the plot in the workflow as well so that the user has a visual reference.

Reply:
Let's include in ESSE schema and make optional? Also, would be good to know which units this is measured in - the 0.3 value. Maybe something to add to the schema field description

Reply:
Per our conversation, we decided to solve this for now by exposing the logic for finding extrema in a python unit.